libs/libcds/docs/index.rst

```diff
@@ -0,0 +1,145 @@
+    Copyright (c) 2020 The STE||AR-Group
+
+    SPDX-License-Identifier: BSL-1.0
+    Distributed under the Boost Software License, Version 1.0. (See accompanying
+    file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+.. _libs_libcds:
+
+======
+libcds
+======
+
+LibCDS Overview
+###############
+
+`LibCDS <https://github.com/khizmax/libcds>`_ implements a collection of
+concurrent containers that don't require external (manual) synchronization
+for shared access, and safe memory reclamation (SMR) algorithms like
+Hazard Pointer and user-space RCU that is used as an epoch-based SMR.
+
+Supporting features
+###################
+Currently, this module **only** supports *Hazard Pointer*
+(a type of safe memory reclamation schemas/garbage collectors)
+based containers in LibCDS
+at HPX light-weight user-level thread. In the future, we might consider
+adding support to other containers, including dynamic Hazard Pointer, RCU, etc,
+at HPX user-level thread,
+and might also explore suitable containers (i.e. flat-combining queue) in HPX
+runtime scheduler.
+
+To find out which container supports Hazard Pointer based garbage collector,
+one might want to check
+`LibCDS Documentation <http://libcds.sourceforge.net/doc/cds-api/index.html>`_.
+For example, after clicking `cds > Modules > FeldmanHashMap <http://libcds.sourceforge.net/doc/cds-api/classcds_1_1container_1_1_feldman_hash_map.html>`_
+, one can find Template parameters in FeldmanHashMap class
+:cpp:class:`cds::container::FeldmanHashMap< GC, Key, T, Traits >`
+suggests *GC - safe memory reclamation schema. Can be gc::HP, gc::DHP or one of RCU type`*
+This means :cpp:type:`FeldmanHashMap` can be safely used with Hazard Pointer GC. However,
+again, current HPX does not support *gc:DHP or RCU*, so we cannot use these two types of garbage collectors.
+
+Build Hazard Pointer w/ HPX threads
+###################################
+To build and your own lock free container in LibCDS using
+HPX light-weight user-level thread, one might first get familiar with
+`LibCDS <https://github.com/khizmax/libcds>`_ itself. The simplest way to
+launch Hazard Pointer in HPX threads is to do the following:
+
+.. code-block:: c++
+
+    #include <hpx/hpx_init.hpp>
+    #include <cds/init.h>       // for cds::Initialize and cds::Terminate
+    #include <cds/gc/hp.h>      // for cds::HP (Hazard Pointer) SMR
+
+    int hpx_main(int, char**)
+    {
+        // Initialize libcds (hazard pointer type by default)
+        hpx::cds::libcds_wrapper cds_init_wrapper;
+
+        {
+            // If main thread uses lock-free containers
+            // the main thread should be attached to libcds infrastructure
+            hpx::cds::hpxthread_manager_wrapper cdswrap;
+
+            // Now you can use HP-based containers in the main thread
+            //...
+        }
+
+        return hpx::finalize();
+    }
+
+    int main(int argc, char* argv[])
+    {
+        return hpx::init(argc, argv);
+    }
```

Comment:
Please move this to the examples directory, and add the file with `literalinclude` instead. Getting the path correct in the `literalinclude` is a bit tricky, so I can help with that part.

Reply:
Can you help with this? I am not sure what `literalinclude` is. Thanks.

Reply:
Yeah, sure. I can push a change for this.

Reply:
I have a commit for this, but it's on another laptop. I'll push it next week.